application/src/main/java/org/thingsboard/server/service/apiusage/BaseApiUsageState.java

```diff
@@ -26,17 +26,19 @@
 import org.thingsboard.server.common.data.id.TenantId;
 import org.thingsboard.server.common.msg.tools.SchedulerUtils;
 
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Map;
-import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
 
 public abstract class BaseApiUsageState {
     private final Map<ApiUsageRecordKey, Long> currentCycleValues = new ConcurrentHashMap<>();
     private final Map<ApiUsageRecordKey, Long> currentHourValues = new ConcurrentHashMap<>();
 
+    private final Map<ApiUsageRecordKey, Map<String, Long>> lastGaugesByServiceId = new HashMap<>();
+    private final Map<ApiUsageRecordKey, Long> gaugesReportCycles = new HashMap<>();
+    private static long gaugeReportInterval = TimeUnit.MINUTES.toMillis(3);
```

Comment:
Please make the report interval configurable via yml (env). You may leave 3 minutes as the default value.